### PR TITLE
complete reviewList, sorting, writeReview components

### PR DESCRIPTION
**Pull request by runhui2010**

### CODE REVIEW

File: MockData/ProductStyles.js
Comment (cschucode) [inline MockData/ProductStyles.js]: Are you using this for reviews?

**Comment by cschucode** [comment]
This looks like it has a bunch of Andy's changes in it. Can you wait until he merges to main and then merge main with your branch?

File: client/components/ReviewRating/ReviewList/Helpfulness.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/Helpfulness.jsx]: This is very minor, but can we import React first as a matter of style and consistency?

File: client/components/ReviewRating/ReviewList/Helpfulness.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/Helpfulness.jsx]: Would this be an `onClickHandler` since it's tied to the onClick event?

File: client/components/ReviewRating/ReviewList/ReviewBody.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/ReviewBody.jsx:43]: This could probably use a line of spacing between proptypes and export statement.

File: client/components/ReviewRating/ReviewList/ReviewList.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/ReviewList.jsx:4]: Can we add a line of spacing here between package and component imports?

File: client/components/ReviewRating/ReviewList/ReviewRating.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/ReviewRating.jsx]: Let's try to be consistent with our imports
* external 
* components
* css

```
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import ReviewList from './ReviewList';
import WriteReview from '../WriteReview/WriteReview';
import Sorting from '../Sorting/Sorting';

import '../../../css/reviewRating.scss';
```

File: client/components/ReviewRating/ReviewList/Helpfulness.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/Helpfulness.jsx]: would we want to fetch reviews inside the 'then' block of this axios put request so as to include the post that was just added?

File: client/components/ReviewRating/ReviewList/StarRating.jsx
Comment (cschucode) [inline client/components/ReviewRating/ReviewList/StarRating.jsx]: We have a component for the now. 😄

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (cschucode) [inline client/components/ReviewRating/WriteReview/WriteReview.jsx]: This is already pretty hard to understand with all the map (i) map (i) etc. Let's not add rules to disable our eslint unless there's no other way around it. If we want to disable the rule, let's discuss and add it to .eslintrc file.

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (cschucode) [inline client/components/ReviewRating/WriteReview/WriteReview.jsx:38]: Looks like a utility function to me. Maybe consider importing from our utils file?

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (cschucode) [inline client/components/ReviewRating/WriteReview/WriteReview.jsx:119]: Maybe create a TODO item to clean this file up. There seems to be quite a bit of repeated code.

File: client/css/reviewRating.scss
Comment (cschucode) [inline client/css/reviewRating.scss]: There should be white space between the colon and the property value.

File: client/tests/ReviewRating.test.jsx
Comment (cschucode) [inline client/tests/ReviewRating.test.jsx:4]: Are we going to use this?

File: server/index.js
Comment (cschucode) [inline server/index.js]: Can we change this endpoint to `/api/reviews/:review_id/helpful` to stay consistent with the others?

File: server/index.js
Comment (cschucode) [inline server/index.js]: Please add `/api` to this endpoint.

**Review by cschucode — CHANGES_REQUESTED** [review]
Overall this looks pretty good. You have a lot of functionality that will require tests as well. I would clean this up as best you can and then create tickets for 
* testing
* refactoring
* clean-up

Also, 15 file PRs take a while, so if we can keep our future PRs to one feature at a time that would be great!

File: client/components/ReviewRating/ReviewList/Helpfulness.jsx
Comment (runhui2010) [reply client/components/ReviewRating/ReviewList/Helpfulness.jsx]: You are right, those are all small things but it is best practice. thanks

File: client/components/ReviewRating/ReviewList/ReviewBody.jsx
Comment (runhui2010) [reply client/components/ReviewRating/ReviewList/ReviewBody.jsx:43]: added.

File: client/components/ReviewRating/ReviewList/ReviewList.jsx
Comment (runhui2010) [reply client/components/ReviewRating/ReviewList/ReviewList.jsx:4]: I will be mindful of the future code.

File: client/components/ReviewRating/ReviewList/Helpfulness.jsx
Comment (runhui2010) [reply client/components/ReviewRating/ReviewList/Helpfulness.jsx]: sure, that might be the reason for the delay. thanks.

File: client/components/ReviewRating/ReviewList/StarRating.jsx
Comment (runhui2010) [reply client/components/ReviewRating/ReviewList/StarRating.jsx]: deleted.

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (runhui2010) [reply client/components/ReviewRating/WriteReview/WriteReview.jsx]: I was struggling with using state on the nested object. That is the way I found to put it into an array so I can map it when the component re-render. I broke down the line so it doesn't break the eslint anymore. thanks

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (runhui2010) [reply client/components/ReviewRating/WriteReview/WriteReview.jsx:38]: That is the function to validate the email input. I don't think other components can apply to it. But if the utility will be a good place to go. I can move it.

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (runhui2010) [reply client/components/ReviewRating/WriteReview/WriteReview.jsx:119]: absolutely, I was trying to get the function to work. Ticket added.

File: client/css/reviewRating.scss
Comment (runhui2010) [reply client/css/reviewRating.scss]: Yes, I wonder if we can install CSS formatter to a quick fix.

File: client/tests/ReviewRating.test.jsx
Comment (runhui2010) [reply client/tests/ReviewRating.test.jsx:4]: We might use it as we will write more tests later on.

File: server/index.js
Comment (runhui2010) [reply server/index.js]: Changed it.

File: server/index.js
Comment (runhui2010) [reply server/index.js]: Changed.

File: client/components/ReviewRating/WriteReview/WriteReview.jsx
Comment (cschucode) [reply client/components/ReviewRating/WriteReview/WriteReview.jsx:38]: It's up to you. Just asking questions for my own clarification. Thanks.

File: client/css/reviewRating.scss
Comment (cschucode) [reply client/css/reviewRating.scss]: I bet one exists, but it will also be able important to see and type code that doesn't look quite right stylistically. It will become second nature after a while.

File: client/tests/ReviewRating.test.jsx
Comment (cschucode) [inline client/tests/ReviewRating.test.jsx:10]: Nice work with the tests! 👍

**Review by cschucode — APPROVED** [review]
Nice work! Sorry to be so meticulous but I promise you'll thank me in the long run. Before you merge this PR, please add a description of the changes... you can list them out in bullet points. It also helps your reviewers know what they're looking at. All PRs should contain a brief description. 🙂

**Pull request merged**
